[wasm] Enable dedup by default.

vargaz opened the pull request.

vargaz committed:
[wasm] Enable dedup by default.

vargaz commented:
/azp run runtime-wasm

radical committed:
[wasm] Enable symbol map for AOT tests too

vargaz commented:
/azp run runtime-wasm

tannergooding commented:
Should be unblocked now. Closing and reopening the PR should retrigger CI with the updated main.

vargaz commented:
/azp run runtime

vargaz commented:
/azp run runtime-wasm

lewing commented:
Looked like the tests failed because the TupleElementNamesAttribute wasn't preserved?

vargaz commented:
/azp run runtime-wasm

vargaz commented:
/azp run runtime-wasm

vargaz commented:
/azp run runtime-wasm

vargaz commented:
The WasmBuildTests failure looks relevant.

vargaz commented:
/azp run runtime-wasm

vargaz commented:
/azp run runtime-wasm

vargaz commented:
/azp run runtime-wasm

vargaz commented:
/azp run runtime-wasm

radical committed:
Merge remote-tracking branch 'origin/main' into wasm-dedup-enable

radical committed:
MonoAOTCompiler: Fix up the path for the output items if they had been

.. copied to `aot-in` for the compilation step.

Example:
- when using WasmDedup=true, we get the main assemblies in `publish`
directory after linking, but `aot-instances.dll` is in different directory.
- this causes `MonoAOTCompiler` to copy all of them to a temporary `aot-in` dir for compiling with `mono-aot-cross`.
- And when the output items are set, we get:

```
 Output Item(s):
     _WasmAssembliesInternal=
         obj/Debug/net8.0/browser-wasm/wasm/for-publish/aot-in/Debug_u4nbxx3i.gc5.dll
                 LlvmBitcodeFile=obj/Debug/net8.0/browser-wasm/wasm/for-publish/Debug_u4nbxx3i.gc5.dll.bc
```

- here the `ItemSpec` is incorrectly set to the temporary `aot-in` path
- which can cause build failures in the following build steps

radical committed:
WBT: remove redundant case

radical commented:
Pushed a fix for the WBT failure.

lewing reviewed (commented):
:shipit:

vargaz commented:
/azp run runtime-wasm

lewing commented:
wbt are green and the build fix makes sense

vargaz commented on eng/testing/tests.wasm.targets:
```diff
@@ -20,7 +20,7 @@
     -->
     <WasmNativeStrip Condition="'$(WasmNativeStrip)' == ''">false</WasmNativeStrip>
-    <WasmEmitSymbolMap Condition="'$(WasmEmitSymbolMap)' == '' and '$(RunAOTCompilation)' != 'true'">true</WasmEmitSymbolMap>
+    <WasmEmitSymbolMap Condition="'$(WasmEmitSymbolMap)' == ''">true</WasmEmitSymbolMap>
```
Should this be reverted from the final PR ?

radical replied:
I had added this because we don't get the symbols on aot now. This lets xharness symbolicate the traces. I think we can keep this.

vargaz commented:
Failures are unrelated.
